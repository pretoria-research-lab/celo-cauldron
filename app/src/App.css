html, body {
  height: 100%;
}

body {
  background-color:var(--celo-white);
}

.App {
  text-align: center;
  background: var(--celo-white);
  font-family: "EB Garamond", "Jost";
  margin: 0px;
  height: 100%;
  display: flex;
  flex-direction: column;
  height: 100vh; /* Avoid the IE 10-11 `min-height` bug. */
}

.App-content {
  flex: 1 0 auto;
}

hr{
  color: var(--celo-dark);
}

.btn-primary {
  background-color: var(--celo-canvas-faint-gold);
  color: var(--celo-dark);
  border: 1px solid var(--celo-dark);
  font-size: 24px;
  margin: 5px;
}

.container {
  display: flex;
  flex-direction: column;
  align-content: center;
}

.column {
  display: flex;
  flex-direction: column;
  align-content: center;
  padding: 10px;
}

.row {  
  display: flex;
  flex-direction: row;
  align-content: center;
  padding: 10px;
}

.centered {
  align-items: center;
}

a {
  color: var(--celo-accent-blue);
}

.page-header {
  align-items: center;
}

.celoIcons {
  max-width: 100px;
}

.jost {
  font-family: Jost;
 }

.header-logo {
  display: flex;
  max-width: 250px;
  max-height: 80%;
}