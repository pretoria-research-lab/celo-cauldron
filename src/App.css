body {
  background-color: black;  
}

.App {
  text-align: center;
  background: black;
  color: var(--terminal);
  font-family: source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace;
  margin:10px;
  border: solid 2px var(--terminal);  
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* h1,h2,h3{
  color: var(--secondary-pelorous);
}
h4,h5{
  color: var(--secondary-pelorous);
} */

.bpf-primary {
  color:  var(--primary-ming);
}
.main-bg {
  background-color: white;
}
.light-highlight-bg {
  background-color: var(--med-gray-1); 
}
.dark-highlight-bg {
  background-color: var(--secondary-vida-loca); 
}

.btn-primary {
  background-color: grey;
  color: var(--terminal);
  border: 1px solid var(--terminal);
  font-size: 24px;
  margin: 5px;
}

.btn-secondary {
  color: var(--terminal);
  border: 1px solid var(--terminal);
  font-size: 24px;
  margin: 5px;
}

.column {
  display: flex;
  flex-direction: column;
  padding: 2px;
}

.loadingImg {
  display: flex;
  max-width: 256px;
}
