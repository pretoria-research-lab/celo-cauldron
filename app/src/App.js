import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import AsyncComponent from "./AsyncComponent";
import Navigation from "./Navigation";
import Footer from "./Footer";
import "./App.css";

// Use AsyncComponent to code split for Javascript load size, only load components on matching route
// See: https://serverless-stack.com/chapters/code-splitting-in-create-react-app.html
// and https://github.com/facebook/create-react-app/blob/master/packages/react-scripts/template/README.md#analyzing-the-bundle-size
// 
const AsyncFaucet = AsyncComponent(() => import("./Faucet"));
const AsyncHome = AsyncComponent(() => import("./Home"));
const AsyncNotFound = AsyncComponent(() => import("./NotFound"));
const AsyncSignedBlocks = AsyncComponent(() => import("./SignedBlocks"));

const faucets = [/*"Alfajores",*/"Baklava"/*, "RC1", "Mainnet"*/];
const signedBlocks = ["RC1"];

class App extends Component {

	render() {

		return (
			<div className="App">				
				<Router> 
					<div className="App-content">         
						<Navigation faucets={faucets} signedBlocks={signedBlocks}/>
						<Switch>
							<Route exact path="/" render={(props) => <AsyncHome {...props} />} />              
							{faucets.map((faucet, i) => <Route key={i} path={"/" + faucet.toLowerCase() + "-faucet"} render={(props) => (<AsyncFaucet key={i} network={faucet} {...props} />)} />)}
							{signedBlocks.map((sb, i) => <Route key={i} path={"/" + sb.toLowerCase() + "-block-map"} render={(props) => (<AsyncSignedBlocks key={i} network={sb} lookback={100} {...props} />)} />)}
							<Route path="*" render={(props) => {return <AsyncNotFound {...props} />; }}/>              
						</Switch>
					</div>
					<Footer faucets={faucets} signedBlocks={signedBlocks}/>
				</Router>						
			</div>   
		);
	}
}

export default App;