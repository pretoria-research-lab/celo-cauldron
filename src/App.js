import React, {Component} from 'react';
import './App.css';
import QRCode from 'qrcode.react';
import loadingImg from './loading.svg';
import axios from 'axios';

const API_CONFIG = {
  baseUrl : "https://xsxgj1rqnc.execute-api.eu-central-1.amazonaws.com",
  basePath : "/dev/reward",
  timeout: 6000,
  headers: {"Content-Type":"application/json;charset=utf-8"}
}

const instance = axios.create({
  baseURL: API_CONFIG.baseUrl,
  timeout: API_CONFIG.timeout,
  headers: API_CONFIG.headers
});

export default class App extends Component {

  constructor(props){
    super(props);
    this.state =  {   claimUrls: [],
                      currentUrl: "",
                      currentUrlIndex: 0,
                      loading: true,
                      reloadTimer: 30
                  };
  }

  getClaimUrls = async () => {
      const response = await instance.get(API_CONFIG.basePath); // no authorization on this endpoint
      console.log("Reward API - getAll() - success, response : " + JSON.stringify(response));
      return response.data.rewards;    
  }

  reloadClaimUrls = async () => {
    this.setState({loading:true}, async () => {
      const claimUrls = await this.getClaimUrls();
      this.setState({claimUrls}, () => {
        this.setState({currentUrl: this.state.claimUrls[this.state.claimUrls.length - 1]}, () => {
          const currentUrlIndex = this.state.claimUrls.length;
          this.setState({currentUrlIndex}, () => {
            this.setState({loading:false});
          });          
        });        
      });
    });
  }

  componentDidMount = () =>{    
    this.reloadClaimUrls();
    setInterval(()=>{
      const reloadTimer =this.state.reloadTimer - 1;
      this.setState({reloadTimer}, async () => {
        if(reloadTimer===0){
          await this.reloadClaimUrls();
          this.setState({reloadTimer:10});
        }
      });      
    }, 1000);
  }

  render = () => {
    return (
      <div className="App">
        <header className="Next Big Thing">          
          <h1>xDAIRow</h1>
          <hr/>
        </header>
        <div id="main" className="container">        
          <div className="row">            
            <div className="column col-sm-12">                
                { this.state.loading ? 
                  <>
                    <h2>Loading...</h2>
                    <div>
                      <img alt="loading" className="loadingImg" src={loadingImg} /> 
                    </div>
                  </>
                  :
                  <>                    
                    <p>Scan this URL and go straight to a browser burner wallet for use in our vending machine!</p>
                    
                    <div className='qrCode'>
                      {this.state.currentUrl.claimUrl ? <QRCode size={256} bgColor="#000000" fgColor="#00FF33" includeMargin={true} value={this.state.currentUrl.claimUrl} />
                      : <h3>:(</h3>}
                    </div>

                    <p><strong>You received {this.state.currentUrl.rewardInXDAI} xDAI</strong></p>
                    <p>{"Attempt : " + this.state.currentUrlIndex}</p>
                    <p>{"Timestamp : " + this.state.currentUrl.createdTimestamp}</p>
                    <p>{"Hash (truncated) : " + this.state.currentUrl.hashCode.substring(0,12)}</p>
                    <p>Refreshing in {this.state.reloadTimer} seconds...</p>
                    
                    <button className="btn-primary" onClick={(event)=>{
                      event.preventDefault();
                      if(this.state.currentUrlIndex!==0){
                        const currentUrlIndex = this.state.currentUrlIndex -1;
                        const currentUrl = this.state.claimUrls[currentUrlIndex];
                        this.setState({currentUrl},()=>this.setState({currentUrlIndex}));  
                      }
                    }}>Show previous...</button>
                    
                    <button className="btn-primary" onClick={(event)=>{
                      event.preventDefault(); 
                      const addToTimer = this.state.reloadTimer + 10; 
                      this.setState({reloadTimer:addToTimer});
                    }}>Wait!</button>
                  </>
                }              
            </div>
          </div> {/* row */}
        </div> {/* main- */}
      </div> 
    );
  }
}
