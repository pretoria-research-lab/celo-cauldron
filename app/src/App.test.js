import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';

it('renders without crashing and matches snapshot', async () => {
  const component = renderer.create(
    <App />,
  );  
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
