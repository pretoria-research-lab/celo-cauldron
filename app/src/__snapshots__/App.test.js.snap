// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing and matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="App-content"
  >
    <nav
      className="navbar navbar-expand-lg navbar-dark static-top"
    >
      <div
        className="container-fluid"
      >
        <a
          className="navbar-brand"
          href="/"
        >
          <img
            alt="Pretoria Monochrome Logo"
            className="header-logo"
            src="Pretoria_Logo_color_02.png"
          />
        </a>
        <button
          aria-controls="navbarResponsive"
          aria-expanded="false"
          aria-label="Toggle navigation"
          className="navbar-toggler"
          data-target="#navbarResponsive"
          data-toggle="collapse"
          type="button"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarResponsive"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <a
                aria-current="page"
                className="nav-link active"
                href="/"
                onClick={[Function]}
                style={Object {}}
              >
                Tools
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/block-map"
                onClick={[Function]}
              >
                Block Map
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/alfajores-faucet"
                onClick={[Function]}
              >
                Alfajores Faucet
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                aria-current={null}
                className="nav-link"
                href="/baklava-faucet"
                onClick={[Function]}
              >
                Baklava Faucet
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
  </div>
  <footer
    className="footer jost"
  >
    <div
      className="container-fluid text-sm-left"
    >
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          <a
            className="navbar-brand"
            href="/"
          >
            <img
              alt="Pretoria Monochrome Logo"
              className="header-logo"
              src="Pretoria_Logo_color_02.png"
            />
          </a>
        </div>
        <div
          className="col-sm-3"
        >
          <p>
            Built by 
            <a
              href="https://pretoriaresearchlab.io"
            >
              Pretoria Research Lab GmbH
            </a>
          </p>
          <p>
            We are a DeFi and staking protocol start-up founded in February 2020 in Berlin, Germany by 
            <a
              href="https://www.nextbigthing.ag/"
            >
              Next Big Thing
            </a>
            , choosing to support the 
            <a
              href="https://celo.org"
            >
              Celo Protocol
            </a>
             with tooling and validating.
          </p>
        </div>
        <div
          className="col-sm-3"
        >
          <p>
            If you find this page useful, please consider supporting Pretoria by voting for our Celo validator group:
          </p>
          <p>
            <a
              href="https://explorer.celo.org/address/0xddaa60b6d803674bbc434f1c2b261ceb67c2fd7c/celo"
            >
              0xddaa60b6d803674bbc434f1c2b261ceb67c2fd7c
            </a>
          </p>
        </div>
        <div
          className="col-sm-3"
        >
          <p>
            Built with 
            <a
              href="https://reactjs.org/"
            >
              React JS
            </a>
            , 
            <a
              href="https://serverless.com/"
            >
              Serverless Framework
            </a>
            , and 
            <a
              href="https://docs.gitlab.com/ee/ci/"
            >
              GitLab CI
            </a>
            . Icons used with limited permission as per the 
            <a
              href="https://celo.org/code-of-conduct"
            >
              Celo Code of Conduct
            </a>
            . Best viewed on desktop.
          </p>
          <p>
            Help us improve this page. Contribute at 
            <a
              href="https://gitlab.com/pretoria-research-lab/celo-cauldron"
            >
              https://gitlab.com/pretoria-research-lab/celo-cauldron
            </a>
          </p>
        </div>
      </div>
    </div>
  </footer>
</div>
`;
